app/services/hearing_recorder.rb

```diff
@@ -8,13 +8,33 @@ def initialize(hearing_id:, body:, publish_to_queue:)
   end
 
   def call
-    hearing.update!(body: body)
-    HearingsCreatorWorker.perform_async(Current.request_id, hearing.id) if publish_to_queue
+    if hearing_contract.success?
+      hearing.update!(body: body)
+      publish_hearing_to_queue
```

Comment:
This is less clear to me now. It looks like it will publish to queue even when `!publish_to_queue`